providers/platform.yaml

```
@@ -0,0 +1,41 @@
+#ddev-generated
+
+auth_command:
+  command: |
+    set -eu -o pipefail
+    if [ -z "${PLATFORMSH_CLI_TOKEN:-}" ]; then echo "Please make sure you have set PLATFORMSH_CLI_TOKEN in ~/.ddev/global_config.yaml" && exit 1; fi
```

I am not sure of the context where this runs. But if it has access to the interactive terminal and the host filesystem then this command would trigger automatic login, and it could account for login via methods other than PLATFORMSH_CLI_TOKEN, and could avoid saving the token in a YAML file:

```suggestion
    if [ -z "$(platform auth:info id >/dev/null)" ]; then echo "Not logged in. Please run: platform login" && exit 1; fi
```

---

Remember that this is inside the web container. I like the technique of checking the auth:info, but in this context they're likely the exact same; there really aren't any other likely ways that anybody would have logged in inside the web container.

---

Yes fair enough.

I'm pretty ignorant of how this works. Is there a context where commands run outside / a way to avoid them running inside the web container? if all else is equal, I'd like to keep CLI use "outside" the container ideally. Local is fine but I don't want to encourage authenticated CLI use inside the p.sh remote container ([reasoning](https://github.com/platformsh/platformsh-cli/blob/7b1dce816cb9de5175f7dde97b44a73b07484467/src/Command/CommandBase.php#L166-L170)).

---

Everything is inside the web container. That means so many things. It means people don't have to install or configure anything. And it works on windows, etc.